Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedMapModel.swift

```diff
@@ -64,16 +122,136 @@ class PreplannedMapModel: ObservableObject, Identifiable {
         }
     }
     
-    /// A Boolean value indicating if download can be called.
-    var canDownload: Bool {
-        switch status {
-        case .notLoaded, .loading, .loadFailure, .packaging, .packageFailure,
-                .downloading, .downloaded:
-            false
-        case .packaged, .downloadFailure:
-            true
+    /// Updates the status based on the download result of the mobile map package.
+    func updateDownloadStatus(for downloadResult: Optional<Result<MobileMapPackage, any Error>>) {
+        switch downloadResult {
+        case .success:
+            status = .downloaded
+            if let mobileMapPackage = try? downloadResult?.get() {
+                self.mobileMapPackage = mobileMapPackage
+            }
+        case .failure(let error):
+            status = .downloadFailure(error)
+        case .none:
+            if let mobileMapPackage = try? downloadResult?.get() {
+                setMobileMapPackage(mobileMapPackage)
+            }
+        }
+    }
+    
+    /// Sets the mobile map package once downloaded.
+    /// - Parameter mobileMapPackage: The mobile map package.
+    func setMobileMapPackage(_ mobileMapPackage: MobileMapPackage) {
+        // Set the mobile map package if already downloaded or the download job succeeded.
+        if job == nil || job?.status == .succeeded {
+            // Set the mobile map package if it downloaded.
+            self.mobileMapPackage = mobileMapPackage
+            status = .downloaded
+        }
+    }
+    
+    /// Sets the mobile map package if downloaded locally.
+    func setMobileMapPackageFromDownloads() {
+        let mobileMapPackages = OfflineMapAreasView.urls(
+            in: preplannedDirectory,
+            withPathExtension: "mmpk"
+        ).map(MobileMapPackage.init(fileURL:))
+        
+        if let mobileMapPackage = mobileMapPackages.first(where: {
+            $0.fileURL.deletingPathExtension().lastPathComponent == preplannedMapArea.id?.rawValue
+        }) {
```

Comment:
Instead of first traversing the directory, create an array of existing `[MobileMapPackage]`, then find the matching package, an alternative can be constructing the URL of the package-in-theory from the `preplannedMapArea.id`, then use `FileManager.fileExists` to test if the file exists.

As the current approach has minimal performance impact (`O(n^2)`), no need to change.

Reply:
Great suggestion, I applied it and removed the `urls` method.